### PR TITLE
feat(dataarts): add resource dataarts studio resource

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:

add resource dataarts studio resource

**Which issue this PR fixes**:
(optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close that issue when PR gets merged)
fixes #xxx

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST="./huaweicloud/services/acceptance/dataarts" TESTARGS="-run TestAccDataArtsStudioResource_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run TestAccDataArtsStudioResource_basic -timeout 360m -parallel 4
=== RUN   TestAccDataArtsStudioResource_basic
=== PAUSE TestAccDataArtsStudioResource_basic
=== CONT  TestAccDataArtsStudioResource_basic
--- PASS: TestAccDataArtsStudioResource_basic (24.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  24.352s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_studio_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+				workspaceID, id)
+		}
+		return fmt.Sprintf("%s/%s", workspaceID, id), nil
+	}
+}
+
+func testAccResource_basic(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_dataarts_studio_resource" "test" {
+  workspace_id = "%s"
```
Comment (jinyangyang222) [inline huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_studio_resource_test.go:124]: workspace_id = "%[1]s"

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_studio_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  name         = "%s"
```
Comment (jinyangyang222) [inline huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_studio_resource_test.go:125]: name         = "%[2]s"

File: huaweicloud/config/endpoints.go [hunk 3]
```diff
@@ -604,6 +605,11 @@ var allServiceCatalog = map[string]ServiceCatalog{
 		Version: "v1",
 		Product: "DataArtsStudio",
 	},
+	"dataartsv1": {
```
Comment (ShiChangkuo) [inline huaweicloud/config/endpoints.go]: please make the key more meaningful.

```suggestion
	"dataarts-dlf": {
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_resource.go [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+			},
+		},
+	}
+}
+
+func resourceStudioResourceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	//nolint:misspell
```
Comment (ShiChangkuo) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_resource.go]: it seems that misspell is unnecessary.

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_resource.go [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+		return diag.Errorf("error creating DataArts Studio resource: %s is not found in API response", "id")
+	}
+	d.SetId(id.(string))
+	return resourceStudioResourceRead(ctx, d, meta)
+}
+
+func buildCreateOrUpdateResourceBodyParams(d *schema.ResourceData) map[string]interface{} {
+	var dependPackages []map[string]string
+
+	if rawParams, ok := d.Get("depend_packages").([]interface{}); ok {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_resource.go]: it's better to use d.GetOk

```suggestion
	if v, ok := d.GetOk("depend_packages"); ok {
		rawParams := v.([]interface{})
```

File: huaweicloud/provider.go [hunk 6]
```diff
@@ -1087,12 +1087,15 @@ func Provider() *schema.Provider {
 			"huaweicloud_modelarts_network":                modelarts.ResourceModelartsNetwork(),
 			"huaweicloud_modelarts_resource_pool":          modelarts.ResourceModelartsResourcePool(),
 
+
```
Comment (ShiChangkuo) [inline huaweicloud/provider.go]:
```suggestion
```

**Pull request merged**